src/datascience-ui/data-explorer/mainPanel.tsx

```diff
@@ -148,11 +154,36 @@ export class MainPanel extends React.Component<IMainPanelProps, IMainPanelState>
                     postOffice={this.postOffice}
                 />
                 {progressBar}
+                {this.renderBreadcrumb()}
                 {this.state.totalRowCount > 0 && this.state.styleReady && this.renderGrid()}
             </div>
         );
     };
 
+    private renderBreadcrumb() {
+        if (this.state.fileName) {
+            let breadcrumbText = this.state.variableName;
+            if (this.state.originalVariableShape) {
+                breadcrumbText += ' (' + this.state.originalVariableShape?.join(', ') + ')';
+            }
+            return (
+                <div className="breadcrumb-container control-container">
```

Comment:
I'd prefer a separate react tsx file for this, perhaps being too much of a purist here, but generally in react, these are treated and craeted as separate components.
FYI - Not a required change.

Reply:
Yeah I didn't think it was necessary because the breadcrumb render code is just 20 lines of code.